lib/aws-xray-sdk/lambda/lambda_emitter.rb

```diff
@@ -0,0 +1,15 @@
+module XRay
+  # LambdaEmitter extends DefaultEmitter to filter out
+  # the subsegments recorded by the AWS Lambda runtime
+  # when the function execution is complete.
+  class LambdaEmitter < XRay::DefaultEmitter
+    def should_send?(entity:)
+      entity.name != '127.0.0.1' #Do not send localhost entities. It's the ruby lambda runtime
```

Comment:
Can you explain what entity is this and why shouldn't it be streamed?

Reply:
When the `net_http` library is instrumented, segments appear in the traces that seem to be the ruby lambda runtime sending back the results of the lambda invocation to the lambda service.

They appear in the x-ray traces like this:
![image](https://user-images.githubusercontent.com/1063621/63632988-67256f00-c684-11e9-8007-03b58e5bc164.png)

When segments from `127.0.0.1` are not sent, the trace looks like:
![image](https://user-images.githubusercontent.com/1063621/63633035-d26f4100-c684-11e9-8638-309bd6de9404.png)

Which seems to be more expected ( ie, user would not expect to see the lambda runtime post in the x-ray trace). The alternative is to modify the instrumentation of the `net_http` library so that it (somehow) does not instrument the AWS lambda runtime calls?

Here's the section of the raw trace that the post appears in:
```
{
            "Document": {
                "id": "321193b7061fddab",
                "name": "127.0.0.1",
                "start_time": 1566623930.7746437,
                "end_time": 1566623930.775297,
                "parent_id": "f594bde1a1f24670",
                "inferred": true,
                "http": {
                    "request": {
                        "url": "http://127.0.0.1:9001/2018-06-01/runtime/invocation/0cf18128-f648-4379-b7fc-2fa4f07d5ed6/response",
                        "method": "POST"
                    },
                    "response": {
                        "status": 202,
                        "content_length": 16
                    }
                },
                "trace_id": "1-5d60c8ba-8227af6a5bfdc58b778dab40"
            },
            "Id": "321193b7061fddab"
        },
```

Reply:
Matching on the `entity.name` seems broad, but there's not much other information in the entity that would help to identify the context it is being sent from.

Reply:
Alternative is to not capture the subsegment by modifying the `net_http` patch around https://github.com/aws/aws-xray-sdk-ruby/blob/master/lib/aws-xray-sdk/facets/net_http.rb#L24 to match requests made by the lambda runtime?

Something like:
```ruby
      LAMBDA_RUNTIME_PATH_PATTERN = /\A\/[0-9-]{10}\/runtime\/invocation\/[0-9a-f-]{36}\/response\Z/.freeze
      def request(req, body = nil, &block)
        if req.path && (req.path == ('/GetSamplingRules') || req.path == ('/SamplingTargets')
                        || req.path =~ LAMBDA_RUNTIME_PATH_PATTERN)
          return super
        end
```

Reply:
My suspicion is that it is because the Ruby runtime is based on the [custom runtime](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-custom.html) work and the [runtime api](https://docs.aws.amazon.com/lambda/latest/dg/runtimes-api.html). 
The other runtimes that were available before Ruby may not use the runtime api?

Reply:
Okay. This could be the reason. I believe we can accept this and not capture the localhost subsegment by using the approach you mentioned in the comment above. Can you make that change as well?